pkg/operator/controller/dns/controller.go

```diff
@@ -846,3 +873,32 @@ func getIbmDNSProvider(dnsConfig *configv1.DNS, creds *corev1.Secret, instanceCR
 		return provider, nil
 	}
 }
+
+// domainIsAlreadyPublishedInZone returns true if the domain name in the provided DNSRecord is already published by
+// another existing dnsRecord.
+func domainIsAlreadyPublishedInZone(ctx context.Context, cache cache.Cache, record *iov1.DNSRecord, zone *configv1.DNSZone) (bool, error) {
+	records := iov1.DNSRecordList{}
+	if err := cache.List(ctx, &records, client.MatchingFields{dnsRecordIndexFieldName: record.Spec.DNSName}); err != nil {
+		return false, err
+	}
+
+	if len(records.Items) == 0 {
+		return false, nil
+	}
+
+	for _, existingRecord := range records.Items {
+		// we only care if the domain name is published by a different record, so ignore the matching record if it
+		// already exists.
+		// TODO: There's got to be a better way to match the same object
```

Comment:
Compare by `UID` instead of name.

Reply:
This is still a relevant comment, but I moved the function, which made github think it's outdated. I'll make this change in my next update